job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
@ajvb Yes it's OK to do this lock hopping thing... unless it's bad if `c.jobs` is modified by some other goroutine concurrently, RIGHT before, or RIGHT after line 127 executes; since it's unlocked it could change from under you.

I don't know the code base well enough to be able to tell whether or not that is a problem, but I am concerned about the number of locks and overall trickiness of all these pieces.